src/cluster_slot.c

```diff
@@ -0,0 +1,244 @@
+}
+
+static void addReplySlotStats(client *c, unsigned char *slots) {
+    int num_slots_assigned = countAssignedSlotsFromSlotsArray(slots);
+    addReplyMapLen(c, num_slots_assigned);
+
+    for (int slot = 0; slot < CLUSTER_SLOTS; slot++) {
+        if (slots[slot]) addReplySingleSlotStat(c, slot);
+    }
+}
+
+static void addReplySortedSlotStats(client *c, sortedSlotStatEntry sorted[], long limit) {
+    int num_slots_assigned = countAssignedSlotsFromCurrentShard();
+    int len = min(limit, num_slots_assigned);
+    addReplyMapLen(c, len);
+
+    for (int i = 0; i < len; i++) {
+        addReplySingleSlotStat(c, sorted[i].slot);
+    }
+}
+
+static void sortAndAddReplySlotStats(client *c, int order_by, long limit, int desc) {
+    sortedSlotStatEntry sorted[CLUSTER_SLOTS];
+    sortSlotStats(sorted, order_by, desc);
+    addReplySortedSlotStats(c, sorted, limit);
+}
+
+void clusterSlotStatsCommand(client *c) {
+    if (server.cluster_enabled == 0) {
+        addReplyError(c,"This instance has cluster support disabled");
+        return;
+    }
+
+    /* Initialize slot assignment array. */
+    unsigned char slots[CLUSTER_SLOTS]= {UNASSIGNED_SLOT};
+
+    /* No further arguments. */
+    if (c->argc == 2) {
+        /* CLUSTER SLOT-STATS */
+        markAssignedSlots(slots);
+        addReplySlotStats(c, slots);
+        return;
+    }
+
+    /* Parse additional arguments. */
+    if (!strcasecmp(c->argv[2]->ptr,"slotsrange") && c->argc == 5) {
+        /* CLUSTER SLOT-STATS SLOTSRANGE start-slot end-slot */
+        int startslot, endslot;
+        if ((startslot = getSlotOrReply(c,c->argv[3])) == C_ERR ||
+            (endslot = getSlotOrReply(c,c->argv[4])) == C_ERR) {
+            return;
+        }
+        if (startslot > endslot) {
+            addReplyErrorFormat(c,"start slot number %d is greater than end slot number %d", startslot, endslot);
+            return;
+        }
+        checkSlotAssignment(slots, startslot, endslot);
+        addReplySlotStats(c, slots);
+    } else if (!strcasecmp(c->argv[2]->ptr,"orderby") && c->argc >= 4) {
+        /* CLUSTER SLOT-STATS ORDERBY column [LIMIT limit] [ASC | DESC] */
+        int desc = 1, order_by = ORDER_BY_INVALID;
+        if (!strcasecmp(c->argv[3]->ptr, "key_count")) {
+            order_by = ORDER_BY_KEY_COUNT;
+        } else {
+            addReplyError(c, "unrecognized sort column for ORDER BY. The supported columns are: key_count.");
+            return;
+        }
+        int i = 4; /* Next argument index, following ORDERBY */
+        int limit_counter = 0, asc_desc_counter = 0;
+        long limit;
+        while(i < c->argc) {
+            int moreargs = c->argc > i+1;
+            if (!strcasecmp(c->argv[i]->ptr,"limit") && moreargs) {
+                if (getRangeLongFromObjectOrReply(
+                    c, c->argv[i+1], 1, CLUSTER_SLOTS, &limit,
+                    "limit has to lie in between 1 and 16384 (maximum number of slots)") != C_OK)
```

Comment:
I think `lie` is one of those words that's ambiguous for non-native speakers because of the dual meaning. Also minor suggestion that the extra information about maximum slots is unnecessary and might cause confusion.
```suggestion
                    "limit must be between 1 and 16384") != C_OK)
```